### PR TITLE
Upgrade scalafmt 6.2

### PR DESCRIPTION
**Pull request by stephennancekivell**
Upgrade to scalafmt 6.2 using the new format-on-compile plugin.

As previously discussed in #212 This is probably the better way to do it.

See this commit for the upgrade the second is just applying formatting from the new defaults, no settings have been changed. https://github.com/stephennancekivell/alpakka/commit/94772436b609bea9a611c4ef19c841d545ec5455

### CODE REVIEW

File: project/plugins.sbt [hunk 1]
```diff
@@ -1,9 +1,10 @@
-addSbtPlugin("de.heikoseeberger"                 % "sbt-header"       % "1.6.0")
-addSbtPlugin("com.geirsson"                      % "sbt-scalafmt"     % "0.4.10")
-addSbtPlugin("com.dwijnand"                      % "sbt-dynver"       % "1.1.1")
-addSbtPlugin("com.lightbend.paradox"             % "sbt-paradox"      % "0.2.9")
-addSbtPlugin("com.eed3si9n"                      % "sbt-unidoc"       % "0.3.3")
+addSbtPlugin("de.heikoseeberger" % "sbt-header" % "1.6.0")
+addSbtPlugin("com.geirsson" % "sbt-scalafmt" % "0.6.3")
```
Comment (hseeberger) [inline project/plugins.sbt]: This needs to be upgraded to 0.6.6

**Comment by hseeberger** [comment]
@patriknw I could take care of this one if you want – I have been fighting with scalafmt for a couple of weeks ...

**Comment by patriknw** [comment]
@hseeberger that would be great, thanks

**Comment by hseeberger** [comment]
@stephennancekivell thanks a lot for you work. scalafmt has seen a lot of changes and a series of releases recently. Therefor this PR has become a little hairy and I have decided to close it and create a new one under https://github.com/akka/alpakka/pull/231. I will mention your work there, of course.

**Comment by stephennancekivell** [comment]
Hi @hseeberger, sure thing. Your right this PR is looking messy. I should have squashed my commits and rebased instead of merging.

Thanks

**Comment by hseeberger** [comment]
Not your fault. Too many changes of scalafmt over the last weeks. Also some issues with the scalafmt settings used in this project. Taking a fresh stab (first updating and changing settings, then formatting) makes things clearer.

**Pull request closed without merging**